File: Unidad 3/Métododeconvolución.py
import numpy as np

def poisson_convolucion(lmbda, n):
    resultados = []
    for _ in range(n):
        L = np.exp(-lmbda)
        k = 0
        p = 1.0
        while p > L:
            k += 1
            p *= np.random.rand()
        resultados.append(k-1)
    return resultados

print(poisson_convolucion(4, 13))

File: Unidad 3/Método de convolución.R
poisson_convolucion <- function(lambda, n) {
  resultados <- numeric(n)
  for (j in 1:n) {
    L <- exp(-lambda)
    k <- 0
    p <- 1
    while (p > L) {
      k <- k + 1
      p <- p * runif(1)
    }
    resultados[j] <- k - 1
  }
  return(resultados)
}

poisson_convolucion(4, 10)
